src/univariate/continuous/loguniform.jl

```diff
@@ -0,0 +1,81 @@
+"""
+    LogUniform(a,b)
+
+A positive random variable `X` is log-uniformly with parameters `a` and `b` if the logarithm of `X` is `Uniform(log(a), log(b))`.
+The *log uniform* distribution is also known as *reciprocal distribution*.
+```julia
+LogUniform(1,10)
+```
+External links
+
+* [Log uniform distribution on Wikipedia](https://en.wikipedia.org/wiki/Reciprocal_distribution)
+"""
+struct LogUniform{T<:Real} <: ContinuousUnivariateDistribution
+    a::T
+    b::T
+    LogUniform{T}(a::T, b::T) where {T <: Real} = new{T}(a, b)
+end
+Base.eltype(::Type{LogUniform{T}}) where {T} = float(T)
+
+function LogUniform(a::T, b::T; check_args=true) where {T <: Real}
+    check_args && @check_args(LogUniform, 0 < a < b)
+    LogUniform{T}(a, b)
+end
+
+function LogUniform(a::Real, b::Real)
+    a1,b1 = promote(float(a), float(b))
+    LogUniform(a1,b2)
+end
```

Comment:
It should not be necessary to convert the parameters to floating point numbers. In fact, for most computations it is easier if they are kept as integers since then no undesired type promotions occur (e.g., `Int` would be changed to `Float64` and thus computations with samples of type `Float32` will be promoted to `Float64`).
```suggestion
LogUniform(a::Real, b::Real) = LogUniform(promote(a, b)...)
```